### DISCUSSION TITLE
I want to be able to set the Role of the Thread Message object.

### DISCUSSION THREAD

**Discussion by deliqs** [RageAgainstThePixel/OpenAI-DotNet#307] [Q&A] [answered]
# Feature Request

## Is your feature request related to a problem? Please describe.
I want to be able to set the Role of a Thread Message object to Assistant. Currently only User is supported, while the API of OpenAI supports setting this value to Assistant as well. See the docs [here](https://platform.openai.com/docs/api-reference/threads/createThread#threads-createthread-messages).

## Describe the solution you'd like
I want the Thread Message object to have a constructor that accepts a Role parameter. or that the Role property has a setter.

## Describe alternatives you've considered
None as there is no way to achieve this.

**Comment by StephenHodgson** [comment, marked as answer]
Most messages you can already do this in constructor.

In V1 Assistants API you could not.
New v2 Assistant API will support this in 8.0.0
